apollo-router/src/axum_factory/utils.rs

```diff
@@ -127,18 +127,18 @@ pub(super) async fn check_accept_header(
 ) -> Result<Response, Response> {
     let ask_for_html = req.method() == Method::GET && prefers_html(req.headers());
 
-    if req.headers().get(ACCEPT).is_some()
-        && (accepts_wildcard(req.headers())
-            || ask_for_html
-            || accepts_multipart(req.headers())
-            || accepts_json(req.headers()))
+    if req.headers().get(ACCEPT).is_none()
```

Comment:
maube the condition here can be removed entirely because `accepts_json` starts with

```rust
!headers.contains_key(ACCEPT)
        || headers.get_all(ACCEPT).iter().any(|value| {
```

Reply:
Not really because it's also used in the execution part.

Reply:
what do you mean?

Reply:
Not the execution part, the handler. Sorry, just [here](https://github.com/apollographql/router/blob/bnjjj/fix_1990/apollo-router/src/axum_factory/handlers.rs#L143:L143)

Reply:
ok I think what we are saying is not clear. I am proposing that you replace 

```rust
if req.headers().get(ACCEPT).is_none()
        || accepts_wildcard(req.headers())
        || ask_for_html
        || accepts_multipart(req.headers())
        || accepts_json(req.headers())
```

with 

```rust
if accepts_wildcard(req.headers())
        || ask_for_html
        || accepts_multipart(req.headers())
        || accepts_json(req.headers())
```

because `accepts_json` will succeed if `!headers.contains_key(ACCEPT)`, and that's the right place where we want to have that condition, because we need to assume that no accept header = `Accept: application/json`

Reply:
Fixed